solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/TupleStream.java

```diff
@@ -141,19 +141,30 @@ public UUID getStreamNodeId(){
       shards = shardsMap.get(collection);
     } else {
       //SolrCloud Sharding
-      CloudSolrClient cloudSolrClient =
-          Optional.ofNullable(streamContext.getSolrClientCache()).orElseGet(SolrClientCache::new).getCloudSolrClient(zkHost);
+      SolrClientCache solrClientCache = (streamContext != null ? streamContext.getSolrClientCache() : null);
```

Comment:
I think this ternary folds into the conditional that we have:

```
      SolrClientCache scc, lscc;
      if (streamContext != null) {
        scc = streamContext.getSolrClientCache();
      } else {
        scc = lscc = new SolrClientCache();
      }
```

Reply:
It's possible for `streamContext.getSolrClientCache()` to return `null` and in that case combining the ternary and the conditional would give a different result. Would it help to have a comment re: that subtlety perhaps?

Reply:
Yea, add a comment. I definitely missed that subtlety.